sled-agent/src/bootstrap/hardware.rs

```diff
@@ -186,6 +191,18 @@ impl HardwareMonitor {
 
         let storage_manager = StorageManager::new(&log).await;
 
+        // If our configuration asks for synthetic zpools, insert them now.
```

Comment:
Now that the bootstrap agent has a dependency on M.2-based storage, we rely on these synthetic zpools being inserted earlier.